agent/engine/task_manager.go

```diff
@@ -250,7 +250,7 @@ func (mtask *managedTask) overseeTask() {
 	}
 	// TODO: make this idempotent on agent restart
 	go mtask.releaseIPInIPAM()
-	mtask.cleanupTask(mtask.cfg.TaskCleanupWaitDuration)
+	mtask.cleanupTask(retry.AddJitter(mtask.cfg.TaskCleanupWaitDuration, mtask.cfg.TaskCleanupWaitDurationJitter))
```

Comment:
I know this is still in draft, but... can we seed `rand`  properly in the init function of the `retry` package? Otherwise all hosts will produce the same sequence of random numbers, which could make the issue persist. 

```go
func init() {
	rand.Seed(time.Now().UnixNano())
}
```

Reply:
we already called rand.Seed here - https://github.com/aws/amazon-ecs-agent/blob/master/agent/agent.go#L26. do we need to call it in another package?

Reply:
Oh, didn't realize we already call it, my bad. We only need to call that once :)